src/player/features/loop/ui/LoopButton.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+import React from "react";
+
+import { observer } from "mobx-react-lite";
+import { PiRepeat } from "react-icons/pi";
+
+import { usePlayer } from "@/player/entities/player";
+import { IconButton } from "@/shared/ui/inputs";
+import { LoopFeature } from "../model/LoopFeature";
+
+export const LoopButton = observer(function LoopButton() {
+  const player = usePlayer();
+  const [loopFeature] = React.useState(() => new LoopFeature({ player }));
+  const [isRepeating, setIsRepeating] = React.useState(false);
+
+  const handleClick = () => {
+    setIsRepeating((prevState) => {
+      const newRepeatingState = !prevState;
+      loopFeature.execute(newRepeatingState);
+
+      return newRepeatingState;
+    });
+  };
+
+  return (
+    <IconButton tooltip="Repeat" size="small" onClick={handleClick}>
```

Comment:
Not sure we need this line `const [isRepeating, setIsRepeating] = React.useState(false);`;

Maybe we can just execute the use case. Then read from `player.isLooping` directly?

Reply:
Yes good point, I use `this.isLooping = !this.isLooping` directly in the loop function in the `HowlerAudioManager`

Reply:
Nice 👌. No more setState, we let MobX doing his magic 🪄